addon/serializers/pouch.js

```diff
@@ -15,5 +21,41 @@ export default DS.RESTSerializer.extend({
     if (!json[key]) {
       json[key] = [];
     }
+  },
+
+  _isAttachment(attribute) {
+    return ['attachment', 'attachments'].indexOf(attribute.type) !== -1;
+  },
+
+  serializeAttribute(snapshot, json, key, attribute) {
+    this._super(snapshot, json, key, attribute);
+    if (this._isAttachment(attribute)) {
+      // if provided, use the mapping provided by `attrs` in the serializer
+      var payloadKey = this._getMappedKey(key, snapshot.type);
+      if (payloadKey === key && this.keyForAttribute) {
+        payloadKey = this.keyForAttribute(key, 'serialize');
+      }
+      // assign any attachments to the attachments property, so that relational-pouch
+      // will put these in the special CouchDB _attachments property
+      // this will conflict with any 'attachments' attr in the model
+      // suggest that #toRawDoc in relational-pouch should allow _attachments to be specified
+      json['attachments'] = Object.assign({}, json['attachments'], json[payloadKey]);
```

Comment:
@nolanlawson if the user has an `attr` called 'attachments' in their model then it will be overwritten here. Can we change #toRawDoc in relational-pouch to allow attachments to be provided using an `_attachments` property?